src/opm/parser/eclipse/EclipseState/Aquifer/NumericalAquifer/SingleNumericalAquifer.cpp

```diff
@@ -52,4 +57,107 @@ namespace Opm {
     size_t SingleNumericalAquifer::numConnections() const {
         return this->connections_.size();
     }
+
+    std::array<std::set<size_t>, 3> SingleNumericalAquifer::transToRemove(const EclipseGrid& grid) const {
+        std::array<std::set<size_t>, 3> trans;
+        for (const auto& cell : this->cells_) {
+            const size_t i = cell.I;
+            const size_t j = cell.J;
+            const size_t k = cell.K;
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::XPlus)) {
+                trans[0].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::XMinus)) {
+                trans[0].insert(grid.getGlobalIndex(i-1, j, k));
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::YPlus)) {
+                trans[1].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::YMinus)) {
+                trans[1].insert(grid.getGlobalIndex(i, j-1, k));
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::ZPlus)) {
+                trans[2].insert(cell.global_index);
+            }
+            if (AquiferHelpers::neighborCellInsideReservoirAndActive(grid, i, j, k, FaceDir::ZMinus)) {
+                trans[2].insert(grid.getGlobalIndex(i, j, k-1));
+            }
+        }
+        return trans;
+    }
+
+    void SingleNumericalAquifer::updateCellProps(const EclipseGrid& grid,
+                                                 std::vector<double>& pore_volume,
+                                                 std::vector<int>& satnum,
+                                                 std::vector<int>& pvtnum) const {
+        for (const auto& cell : this->cells_) {
+            const size_t activel_index =  grid.activeIndex(cell.global_index);
```

Comment:
I have understood it so that the aquifercells can be in cells which are originally inactive - then
```
active1_index =  grid.activeIndex( cell.global_index );
```
will fail hard. If need to activate cells due to AQUIFER that is a complex operation and it needs to be implemented in `FieldProps.cpp`.

Reply:
I think we make sure the aquifer cells will be active through https://github.com/OPM/opm-common/pull/2105 .

Reply:
Well #2105 will activate a cell in `EclipseGrid` - but more imortant is the `FieldProps` which is not yet handled.

Reply:
I could not find how to copy the link to the comment. Basically, in `src/opm/parser/eclipse/EclipseState/EclipseState.cpp`, I made the application of numerical aquifer before we called the function `actnum()` of FieldProps.

Reply:
And before the function `actnum() ` is called, the cells have not been deactivated due to FieldProps. Then those aquifer cells will be alive before we call `actnum()` of FieldProps. 

 Then hopefully we update the FieldProps properly, when `actnum()` is called later, they will not deactivate those aquifer cells.